spec/factories/articles.rb

```diff
@@ -5,13 +5,14 @@
     published_at { Time.current }
 
     transient do
-      title { generate :title }
+      title { generate(:title) }
       published { true }
       date { "01/01/2015" }
       tags { "javascript, html, discuss" }
       canonical_url { Faker::Internet.url }
       with_canonical_url { false }
       with_main_image { true }
+      main_image_from_frontmatter { false }
```

Comment:
This change is possibly unnecessary now, but was something I discovered in the process. There's a bit of a race condition, or test order bug a couple lines down, where this attribute is referenced inside the HEREDOC article frontmatter. 

The HEREDOC in the factory references `main_image_from_frontmatter`. When this is used by `create :article` (or `build :article`), it's fine; but when this is used by `attributes_for :article` it will **fail**, but _only_ if `attributes_for :article` is called before `create` or `build`.

After [a bit of investigation](https://gist.github.com/jaw6/d299e3904b6c8fe0bd2de298c127c4da), I'm convinced that this behavior has been here for some time and that we just haven't run the tests "out of order" very often. (And/or this is possibly another of our rogue flaky tests?)

This is almost certainly a bug in FactoryBot, but it looks like explicitly defining this attribute in the factory fixes it. The factory is overdue for a general refactoring, according to both the comment in the file and #18641.

Reply:
🤦 I linked the wrong issue somehow, but that's the one I meant. Thank you!